metadata-ingestion/src/datahub/ingestion/source/lookml.py

```diff
@@ -0,0 +1,578 @@
+        actor = self.source_config.actor
+        sys_time = int(time.time()) * 1000
+
+        dataset_snapshot = DatasetSnapshot(
+            urn=f"urn:li:dataset:(urn:li:dataPlatform:{self.source_config.platform_name}, {dataset_name}, {self.source_config.env})",
+            aspects=[],  # we append to this list later on
+        )
+        dataset_snapshot.aspects.append(Status(removed=False))
+        dataset_snapshot.aspects.append(
+            self._get_upsteam_lineage(looker_view, actor, sys_time)
+        )
+        dataset_snapshot.aspects.append(self._get_schema(looker_view, actor, sys_time))
+
+        mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+
+        return mce
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        viewfile_loader = LookerViewFileLoader(self.source_config.base_folder)
+
+        model_files = sorted(
+            f
+            for f in glob.glob(
+                f"{self.source_config.base_folder}/**/*.model.lkml", recursive=True
+            )
+        )
+        for file_path in model_files:
+            model_name = Path(file_path).stem
+            if not self.source_config.model_pattern.allowed(model_name):
+                self.reporter.report_models_dropped(model_name)
+                continue
+            try:
+                model = self._load_model(file_path)
+            except Exception:
+                self.reporter.report_warning(
+                    "LookML", f"unable to parse Looker model: {file_path}"
+                )
+                continue
+
+            for include in model.resolved_includes:
+                is_view_seen = viewfile_loader.is_view_seen(include)
+                if is_view_seen:
+                    continue
+                looker_viewfile = viewfile_loader.load_viewfile(
+                    include, model.connection
+                )
+                if looker_viewfile is not None:
+                    for raw_view in looker_viewfile.views:
+                        maybe_looker_view = LookerView.from_looker_dict(
+                            raw_view,
+                            model.connection,
+                            looker_viewfile,
+                            viewfile_loader,
+                            self.source_config.parse_table_names_from_sql,
+                        )
+                        if maybe_looker_view:
+                            if self.source_config.view_pattern.allowed(
+                                maybe_looker_view.view_name
+                            ):
+                                mce = self._build_dataset_mce(maybe_looker_view)
+                                workunit = MetadataWorkUnit(
+                                    id=f"lookml-{maybe_looker_view.view_name}", mce=mce
+                                )
+                                self.reporter.report_workunit(workunit)
+                                self.reporter.report_views_scanned()
```

Comment:
`report_workunit` already keeps a counter internally - did you mean for `report_views_scanned` to be outside this `if` statement?

Reply:
Yes! move this, and the other similar calls around a bit.